src/Microsoft.DotNet.VersionTools/Dependencies/MSBuildPackageReferenceUpdater.cs

```diff
@@ -0,0 +1,187 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.VersionTools.Util;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.VersionTools.Dependencies
+{
+    public class MSBuildPackageReferenceUpdater : IDependencyUpdater
+    {
+        public IEnumerable<string> MSBuildFilePaths { get; }
+
+        public MSBuildPackageReferenceUpdater(IEnumerable<string> msbuildFilePaths)
+        {
+            MSBuildFilePaths = msbuildFilePaths;
+        }
+
+        public IEnumerable<DependencyUpdateTask> GetUpdateTasks(IEnumerable<DependencyBuildInfo> dependencyBuildInfos)
+        {
+            var tasks = new List<DependencyUpdateTask>();
+            foreach (string filePath in MSBuildFilePaths)
+            {
+                try
+                {
+                    IEnumerable<DependencyChange> dependencyChanges = null;
+
+                    Action update = FileUtils.GetUpdateFileContentsTask(
+                        filePath,
+                        contents => ReplaceAllDependencyVersions(
+                            contents,
+                            filePath,
+                            dependencyBuildInfos,
+                            out dependencyChanges));
+
+                    // The output json may be different even if there weren't any changes made.
+                    if (update != null && dependencyChanges.Any())
+                    {
+                        tasks.Add(new DependencyUpdateTask(
+                            update,
+                            dependencyChanges.Select(change => change.BuildInfo),
+                            dependencyChanges.Select(change => $"In '{filePath}', {change.ToString()}")));
+                    }
+                }
+                catch (Exception e)
+                {
+                    Trace.TraceWarning($"Non-fatal exception occurred processing '{filePath}'. Skipping file. Exception: {e}. ");
+                }
+            }
+            return tasks;
+        }
+
+        private string ReplaceAllDependencyVersions(
+            string input,
+            string projectJsonFile,
+            IEnumerable<DependencyBuildInfo> buildInfos,
+            out IEnumerable<DependencyChange> dependencyChanges)
+        {
+            XDocument documentRoot = XDocument.Parse(input);
+
+            dependencyChanges = FindAllDependencyProperties(documentRoot)
+                    .Select(dependencyProperty => ReplaceDependencyVersion(projectJsonFile, dependencyProperty, buildInfos))
+                    .Where(buildInfo => buildInfo != null)
+                    .ToArray();
+
+            StringWriter sw = new StringWriter();
+            documentRoot.Save(sw, SaveOptions.None);
```

Comment:
I'll try that out. It might just be that the files I tested were already formatted the same way as this gives me, because I didn't see any extra diffs.